packages/block-library/src/search/edit.js

```diff
@@ -149,21 +151,30 @@ export default function SearchEdit( {
 		);
 	};
 
+	const getButtonClassNames = () => {
+		return classnames( 'wp-block-search__button', colorProps.className );
+	};
+
+	const getButtonStyles = () => ( {
+		...colorProps.style,
+		borderRadius,
+	} );
+
 	const renderButton = () => {
 		return (
 			<>
 				{ buttonUseIcon && (
 					<Button
 						icon={ search }
-						className="wp-block-search__button"
-						style={ { borderRadius } }
+						className={ getButtonClassNames() }
```

Comment:
Why do we apply classes and styles for the colors both to the wrapper and to the richtext?

Reply:
Here I don't apply classes and styles to the wrapper, just to the button -- but it is applied to both the icon-only button and to the richtext, so that colors display correctly for both the text and the icon-only button.